app/api/access_codes.py

```diff
@@ -36,70 +59,37 @@ def query(self, view_kwargs):
             query_ = AccessCode.query.filter(AccessCode.tickets.any(id=ticket.id))
         return query_
 
-    def before_post(self, args, kwargs, data):
-        """
-        method to add user_id to view_kwargs before post
-        :param args:
-        :param kwargs:
-        :param data:
-        :return:
-        """
-        kwargs['user_id'] = current_identity.id
-
-    def before_create_object(self, data, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            data['event_id'] = ticket.event_id
-        if view_kwargs.get('event_id'):
-            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-            data['event_id'] = event.id
-        elif view_kwargs.get('event_identifier'):
-            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
-            data['event_id'] = event.id
-        data['user_id'] = current_identity.id
-
-    def after_create_object(self, obj, data, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            ticket.access_codes.append(obj)
-            self.session.commit()
-
     view_kwargs = True
-    decorators = (jwt_required, )
+    decorators = (api.has_permission('is_coorganizer', fetch='event_id',
+                  fetch_as="event_id", model=AccessCode,
+                  leave_if=lambda a: a.get('user_id') or a.get('ticket_id')), )
```

Comment:
Not sure if this the right way to filter only access-codes for which co-orga access is present. (event_id is required relationship, so it will always be there for an access-code). Please check.